container-images/build_llama_and_whisper.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+# Bash does not easily pass arrays as a single arg to a function.  So, make this a global var in the script.
+CMAKE_FLAGS=""
+LLAMA_CPP_CMAKE_FLAGS=""
+WHISPER_CPP_CMAKE_FLAGS=""
+
+function cmakeCheckWarnings() {
+  awk -v rc=0 '/CMake Warning:/ { rc=1 } 1; END {exit rc}'
+}
+
+function cloneAndBuild() {
+  local git_repo=${1}
+  local git_sha=${2}
+  local install_prefix=${3}
+  local work_dir=$(mktemp -d)
+
+  git clone ${git_repo} ${work_dir}
+  cd ${work_dir}
+  git submodule update --init --recursive
+  git reset --hard ${git_sha}
+  cmake -B build ${CMAKE_FLAGS[@]} 2>&1 | cmakeCheckWarnings
+  cmake --build build --config Release -j$(nproc) -v 2>&1 | cmakeCheckWarnings
+  cmake --install build --prefix ${install_prefix} 2>&1 | cmakeCheckWarnings
+  cd -
+  rm -rf ${work_dir}
+}
+
+function dnfPrepUbi() {
+
+  local url="https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm"
+  local uname_m=$(uname -m)
+
+  dnf install -y ${url}
+  crb enable # this is in epel-release, can only install epel-release via url
+  dnf copr enable -y slp/mesa-krunkit epel-9-${uname_m}
+  url="https://mirror.stream.centos.org/9-stream/AppStream/${uname_m}/os/"
+  dnf config-manager --add-repo ${url}
+  url="http://mirror.centos.org/centos/RPM-GPG-KEY-CentOS-Official"
+  curl --retry 8 --retry-all-errors -o /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Official ${url}
+  rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Official
+}
+
+function main() {
+  set -ex
+
+  local container_image=${1}
+  local install_prefix=/usr
+  local package_list
+  local whisper_cpp_sha=${WHISPER_CPP_SHA:-8a9ad7844d6e2a10cddf4b92de4089d7ac2b14a9}
+  local llama_cpp_sha=${LLAMA_CPP_SHA:-aa6fb1321333fae8853d0cdc26bcb5d438e650a1}
+  local common_rpms=("python3" "python3-pip" "python3-argcomplete" "python3-dnf-plugin-versionlock" "gcc-c++" "cmake" "vim" "procps-ng" "git" "dnf-plugins-core" "libcurl-devel")
+  local vulkan_rpms=("vulkan-headers" "vulkan-loader-devel" "vulkan-tools" "spirv-tools" "glslc" "glslang")
+  local intel_rpms=("intel-oneapi-mkl-sycl-devel" "intel-oneapi-dnnl-devel" "intel-oneapi-compiler-dpcpp-cpp" "intel-level-zero" "oneapi-level-zero" "oneapi-level-zero-devel" "intel-compute-runtime")
+
+  LLAMA_CPP_CMAKE_FLAGS+=("-DGGML_CCACHE=OFF" "-DGGML_NATIVE=OFF" "-DBUILD_SHARED_LIBS=NO" "-DLLAMA_CURL=ON")
```

Comment:
This isn't right, we want BUILD_SHARE_LIBS for one but not the other

Reply:
I think -DBUILD_SHARED_LIBS breaks for the intel-gpu build.  I'll double check that.

Reply:
I bet that's why the rocm image got really big...  `-DBUILD_SHARED_LIBS=NO`.  That resulted in some huge executables in /usr/bin...

Reply:
Yeah, I think you got it, -DBUILD_SHARED_LIBS=NO end up in the libraries being duplicated in every single binary produced.

The reason we do it for whisper.cpp is it uses some of the same libraries, with the same names, but with different versions sometimes, so one install will replace the library of the other, breaking whichever was installed first.

whisper.cpp is a small project so we statically link that one to fix the issue.